base/libc.jl

```diff
@@ -77,6 +77,18 @@ flush_cstdio() = ccall(:jl_flush_cstdio, Void, ())
 @unix_only systemsleep(s::Real) = ccall(:usleep, Int32, (UInt32,), round(UInt32,s*1e6))
 @windows_only systemsleep(s::Real) = (ccall(:Sleep, stdcall, Void, (UInt32,), round(UInt32,s*1e3)); return Int32(0))
 
+type TimevalStruct
+   sec::Clong   # time_t
+   usec::Clong  # suseconds_t
+end
+
+function TimevalStruct()
+    tv = TimevalStruct(0,0)
+    success = ccall(:gettimeofday, Cint, (Ref{TimevalStruct}, Ref{Void}), tv, C_NULL) == 0
+    success || error("unable to determine current time")
+    return tv
+end
```

Comment:
Should be `tv = Ref{TimevalStruct}()` followed by `return tv[]`.  Also, `Ptr{Void}`, not `Ref{Void}`.